src/main/java/com/iridium/iridiumteams/commands/TopCommand.java

```diff
@@ -1,26 +1,130 @@
 package com.iridium.iridiumteams.commands;
 
+import com.iridium.iridiumcore.utils.StringUtils;
 import com.iridium.iridiumteams.IridiumTeams;
+import com.iridium.iridiumteams.Rank;
 import com.iridium.iridiumteams.database.IridiumUser;
 import com.iridium.iridiumteams.database.Team;
 import com.iridium.iridiumteams.gui.TopGUI;
+import com.iridium.iridiumteams.sorting.TeamSorting;
 import lombok.NoArgsConstructor;
+import org.bukkit.command.CommandSender;
 import org.bukkit.entity.Player;
 
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
+import java.util.stream.Collectors;
 
 @NoArgsConstructor
 public class TopCommand<T extends Team, U extends IridiumUser<T>> extends Command<T, U> {
 
-    public TopCommand(List<String> args, String description, String syntax, String permission, long cooldownInSeconds) {
+    public String adminPermission;
+
+    public TopCommand(List<String> args, String description, String syntax, String permission, long cooldownInSeconds, String adminPermission) {
         super(args, description, syntax, permission, cooldownInSeconds);
+        this.adminPermission = adminPermission;
     }
 
     @Override
-    public boolean execute(U user, String[] arguments, IridiumTeams<T, U> iridiumTeams) {
-        Player player = user.getPlayer();
-        player.openInventory(new TopGUI<>(iridiumTeams.getTop().valueTeamSort, player, iridiumTeams).getInventory());
-        return true;
+    public boolean execute(CommandSender sender, String[] arguments, IridiumTeams<T, U> iridiumTeams) {
+
+        int listLength = 10;
+        boolean showValue = true;
+        boolean excludePrivate = !sender.hasPermission(adminPermission);
+
+        if (sender instanceof Player && arguments.length == 0) return sendGUI((Player) sender, iridiumTeams);
+
+        switch (arguments.length) {
+            case 3: {
+                try {
+                    listLength = Integer.parseInt(arguments[2]);
+                } catch (NumberFormatException ignored) {}
+            }
+            case 2: {
+                if (arguments[1].equalsIgnoreCase("experience")) showValue = false;
```

Comment:
id construct this dynamically, add a name value to the TeamSorting class, and select that via the argument